docs/design/password-key-derivation.md

```diff
@@ -8,35 +8,41 @@ This design doc describes the way cryptographic encryption keys will be derived
 
 A unique encryption key exists for every path inside of a bucket, and they are designed in a way that allows for deriving a key from a parent and some path component. For example, if you had the key for `sj://bucket/path1/path2` then you could derive the key for `sj://bucket/path1/path2/path3`.
 
-That leaves open the question for how a root key is derived. Some requirements on that root key derivation include
+That leaves open the question for how a root key is created. Some requirements on that root key creation include
 
 1. It's based on the user's choice (for example, a password).
-2. If the user inputs the same password on different machines, the same root key should be derived.
+2. If the user inputs the same password on different machines, the same root key should be created.
 
 These requirements allow users to be in full control of their encryption, and don't require users safely transporting high entropy (hard to remember) secrets to bootstrap new uplinks.
 
 This design accomodates more requirements that allow for additional features:
 
-3. A root key can be derived for any path in a bucket.
-4. Buckets root keys should not depend on the name of the bucket.
-5. A table of keys for low entropy passwords should not be possible. In other words, an attacker with knowledge of the algorithm should not be able to use a dictionary of common passwords and pre-compute what keys to check in the event of a data breach.
+3. A root key can be created for any encrypted path in a bucket, not just the bucket.
+4. Root keys should not depend on the name of the bucket.
+5. A table of root keys for low entropy passwords should not be possible. In other words, an attacker with knowledge of the algorithm should not be able to use a dictionary of common passwords and pre-compute what keys to check in the event of a data breach.
 6. Users do not have to enter a password for every bucket they create: they can have a default password that is used for every bucket unless otherwise specified.
 
 The third requirement allows having multiple encrypted domains to exist within a single bucket, allowing delegation of encryption. The fourth requirement allows renaming buckets without having to re-encrypt all of the paths inside of a bucket. The fifth requirement enhances security if a satellite breach happens. This is usually accomplished with [salting](https://en.wikipedia.org/wiki/Salt_(cryptography)). The sixth requirement allows providing a finite amount of information to a third party that allows them access to the keys for any bucket created with the default password in the future.
 
 ## Design
 
-Satellites will add a `salt` field to bucket metadata that lives with the path encryption information that already exists and that we already request. Satellites will also store a `salt` that lives with the project metadata. The salts will be 32 random, stable bytes.
+First, a **root key** is defined to be a key for some bucket and encrypted path.
```

Comment:
I really don't want to quibble any more about terminology but this seems like it might be very misleading. We're not saying that _any_ key for a (bucket, encryptedPath) pair is a "root key", are we? I thought we were using "root key" to mean a key that isn't derived using the typical `hmac(parentKey, pathComponent)` approach.

Reply:
This is a fine quibble. 👄👌 . I'm trying to say that for any (bucket, encryptedPath) pair, any number of root keys can be created, and you can derive keys from any root key.

I'm having a hard time coming up with clear, unambiguous, and concise phrasing.

Reply:
Maybe the equation on line 33 is actually the definition of a root key? I read it as just saying that it might not match the derived key, but if it's the definition then this all seems fine